mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -160,10 +160,17 @@ GemmRewritePattern::matchAndRewrite(GemmOp op, GemmOpAdaptor adaptor,
 
   auto accumulator = getAccumulator(a, b, c, rw, loc);
   if (isAccel) {
-    rw.create<GridwiseGemmAccelOp>(
+    auto gridwieseGemmOp = rw.create<GridwiseGemmAccelOp>(
         loc, a, b, accumulator, op.getArchAttr(), numCUAttr,
         op.getFeaturesAttr(), op.getStoreMethodAttr(), blockSize, gridSize,
         params.cast<RockAccelTuningParamAttrInterface>());
```

Comment:
I would try to hook it from here and not touch `GridwiseToBlockwise`. Given that `a=[g,m,k], b=[g,k,n], c = [g,m,n]` What I would do is:
```
asplit = transform.slice(a, splitk, bidx, tidx) : <gxmxk> -> <gxmxsplitk>
bsplit = transform.slice(b, splitk,bidx,tidx) : <gxmxk> -> <gxsplitkxn>
csplit = transform.slice(c, splitk, bidx, tidx) : <gxmxk> -> <gxmxn>

if (bidx==0){
    cGlobal = atomicAdd(cGlobal, csplit)
}
```

Reply:
Hi @giuseros! I'd like to continue our discussion about Split-K. Honestly, I didn't get how `slice` is supposed to help in this case. Would you mind to schedule one to one meeting and discuss it again?